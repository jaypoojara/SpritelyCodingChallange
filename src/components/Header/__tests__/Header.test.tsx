import React from 'react';
import renderer from 'react-test-renderer';
import Headers from '../index';
import {render, fireEvent, screen} from '@testing-library/react-native';

describe('<Header />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Headers />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('will hide notifications if notification count is 0', () => {
    const onPressMock = jest.fn();
    render(<Headers notificationCount={0} onPressNotification={onPressMock} />);
    const element = screen.queryByTestId('notification-count-id');
    expect(element).toBeFalsy();
  });

  it('will render notifications', () => {
    const onPressMock = jest.fn();
    render(<Headers notificationCount={2} onPressNotification={onPressMock} />);
    const element = screen.getByTestId('notification-count-id');
    expect(element).toBeTruthy();
  });

  it('gives callback when notification icon is pressed', async () => {
    const mockOnPress = jest.fn();
    render(<Headers notificationCount={0} onPressNotification={mockOnPress} />);
    const element = screen.getByTestId('notification-button-id');
    fireEvent.press(element);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
