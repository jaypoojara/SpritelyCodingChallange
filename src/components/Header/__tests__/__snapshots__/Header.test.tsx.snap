// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
  testID="home-header-id"
>
  <Image
    source={
      Object {
        "testUri": "../../../src/asset/icons/spritely_logo.png",
      }
    }
    style={
      Object {
        "resizeMode": "contain",
        "width": 110,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 25,
        "elevation": 5,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 10,
        "width": 50,
      }
    }
    testID="notification-button-id"
  >
    <View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#003A67",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰂚
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
  testID="home-header-id"
>
  <Image
    source={
      Object {
        "testUri": "../../../src/asset/icons/spritely_logo.png",
      }
    }
    style={
      Object {
        "resizeMode": "contain",
        "width": 110,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 25,
        "elevation": 5,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 10,
        "width": 50,
      }
    }
    testID="notification-button-id"
  >
    <View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#003A67",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰂚
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
